### PR TITLE
chore: split omnibar state out of store

### PR DESCRIPTION
**Pull request by johnkim-det**
## Description

[WEB-588](https://determinedai.atlassian.net/browse/WEB-588)

## Test Plan

Omnibar should still function as expected (toggle on Ctrl + Space, hide on Esc, commands work)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/omnibar/Omnibar.tsx [hunk 1]
```diff
@@ -19,12 +19,11 @@ const omnibarInput = () =>
   document.querySelector('#omnibar input[type="text"]') as HTMLInputElement | null;
 
 const Omnibar: React.FC = () => {
```
Comment (ClaireNeveu) [inline webui/react/src/omnibar/Omnibar.tsx:21]: Is there a reason the omnibar state can't just be local state in this component? Does the keydown listener have to be in `App`?

Comment (johnkim-det) [reply webui/react/src/omnibar/Omnibar.tsx:21]: definitely works fine when it's just local state here. 

my guess is that it was placed in `App` alongside `useKeyTracker`, but there's really no need to have both there, supported by Context. will update this PR to remove `OmnibarContext` and `OmnibarProvider` and use local state instead.

File: webui/react/src/omnibar/Omnibar.tsx [hunk 2]
```diff
@@ -19,12 +19,27 @@ const omnibarInput = () =>
...
-  const storeDispatch = useStoreDispatch();
-  const { ui } = useStore();
+  const [showing, setShowing] = useState(false);
+
+  useEffect(() => {
+    const keyDownListener = (e: KeyboardEvent) => {
+      if (e.code === KeyCode.Space && e.ctrlKey) {
+        setShowing(!showing);
```
Comment (trentwatt) [inline webui/react/src/omnibar/Omnibar.tsx:27]:
```suggestion
        setShowing((showing) => !showing);
```

File: webui/react/src/omnibar/Omnibar.tsx [hunk 3]
```diff
@@ -19,12 +19,27 @@ const omnibarInput = () =>
...
+        setShowing(false);
+      }
+    };
+
+    keyEmitter.on(KeyEvent.KeyDown, keyDownListener);
+
+    return () => {
+      keyEmitter.off(KeyEvent.KeyDown, keyDownListener);
+    };
+  }, [showing, setShowing]);
```
Comment (trentwatt) [inline webui/react/src/omnibar/Omnibar.tsx:38]:
```suggestion
  }, []);
```

Comment (trentwatt) [reply webui/react/src/omnibar/Omnibar.tsx:38]: react says you dont need to put `setState`in deps

File: webui/react/src/omnibar/Omnibar.tsx [hunk 4]
```diff
@@ -19,12 +19,27 @@ const omnibarInput = () =>
...
 
   const hideBar = useCallback(() => {
-    storeDispatch({ type: StoreAction.HideOmnibar });
-  }, [storeDispatch]);
+    setShowing(false);
+  }, [setShowing]);
```
Comment (trentwatt) [inline webui/react/src/omnibar/Omnibar.tsx:42]: likewise

**Review by trentwatt — APPROVED** [review]
LGTM, minor nits. using callback form of `setState` means you don't have to attach and detach the listeners every time viz is toggled.

also FWIW: this is not particular to this PR but it looks like if you type something that doesn't match an omnibar command, you get a "Failed to Query" error popup. IMO a "no matches" situation shouldn't be presented as an error.

**Review by ClaireNeveu — APPROVED** [review]

**Pull request merged**